### PR TITLE
Feature-implement-mission-joining

### PR DESCRIPTION
**Pull request by maximoortelli**
### Feature-implement-mission-joining:

**Requirements:**

- _When a user clicks the "Join Mission" button, action needs to be dispatched to update the store. You need to get the ID of the selected mission and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all missions, but the selected mission will have an extra key `reserved` with its value set to `true`. You could use a JS `filter()` or `map()` to set the value of the new state - i.e.:_
```javascript
const newState = state.map(rocket => {
    if(mission.id !== id) 
        return mission;
    return { ...mission, reserved: true };
});

```

- _Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument._

- _Follow the same logic as with the "Join mission" - but you need to set the reserved key to false._

- _Dispatch these actions upon click on the corresponding buttons._

- Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).

Use the React conditional rendering syntax:
```javascript
{rocket.reserved && ( 
    // render Cancel Rocket button
)}
```

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -21,7 +22,14 @@ const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
...
   initialState,
-  reducers: {},
+  reducers: {
+    member: (state, action) => {
+      const missionId = action.payload;
+      state.missions = state.missions.map((mission) => (mission.mission_id === missionId
+        ? { ...mission, activeMember: !mission?.activeMember ?? true }
+        : mission));
+    },
+  },
```
Comment (Estete9) [inline src/redux/missions/missionsSlice.js:32]: You did great with this logic, the use of the terenary function ( ? : ) is great! good job 😄
Please help me change the key from `activeMember` to `reserved`

It's ok that you use this name in all other instances but the Requirements ask for the key `reserved`

**Review by Estete9 — CHANGES_REQUESTED** [review]
Hi @maximoortelli 
I'll review this PR. ✋

Good job so far it looks quite good! That was very close!




There are some recommendations I would like you to take into account.


Highlights
 You have written a clean code ✅
 The distribution, padding, and margins are perfect. ✅
 Good PR title/description ✅
Required Changes ♻️
 Good job adding the elements in the list in the main section, but we can improve it a little bit more together. 💪 
 
Check the comments under the review.

Cheers and Happy coding!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/esteban-palacios-5030a772/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @NaughTban
</a>

</td>
</tr>

File: src/redux/missions/missionsSlice.js (hunk 1 above)
Comment (maximoortelli) [reply src/redux/missions/missionsSlice.js:32]: Done. thank you @Estete9 👍😄

**Review by Estete9 — APPROVED** [review]
## Hi @maximoortelli , 

Your feature is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Great use of modules ✅ 
- Linters are green ✅ 
- Meaningful **Github** commits ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/esteban-palacios-5030a772/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @NaughTban
</a>

</td>
</tr>
</table>

**Pull request merged**